src/v2.rs

```diff
@@ -169,12 +178,141 @@ impl EszipV2Modules {
   }
 }
 
+#[derive(Debug, Clone, Copy)]
+struct Options {
+  /// Hash Function used to checksum the contents of the eszip when encoding/decoding
+  ///
+  /// If the eszip does not include the option, it defaults to `[Checksum::NoChecksum]` in >=v2.2
+  /// and `[Checksum::Sha256]` in older versions.  It is `None` when the eszip header includes a
+  /// checksum that this version of the library does not know.
+  checksum: Option<Checksum>,
+
+  /// Size in Bytes of the hash function digest.
+  ///
+  /// Defaults to the known length of the configured hash function. Useful in order to ensure forwards compatibility,
+  /// otherwise the parser does not know how many bytes to read.
+  checksum_size: Option<u8>,
+}
+
+impl Options {
+  fn default_for_version(magic: &[u8; 8]) -> Self {
+    let defaults = Self {
+      checksum: Some(Checksum::NoChecksum),
+      checksum_size: Default::default(),
+    };
+    #[cfg(feature = "sha256")]
+    let mut defaults = defaults;
+    if let ESZIP_V2_MAGIC | ESZIP_V2_1_MAGIC = magic {
+      // versions prior to v2.2 default to checksuming with SHA256
+      #[cfg(feature = "sha256")]
+      {
+        defaults.checksum = Some(Checksum::Sha256);
+      }
+    }
+    defaults
+  }
+}
+
+impl Default for Options {
+  fn default() -> Self {
+    Self::default_for_version(LATEST_VERSION)
+  }
+}
+
+impl Options {
+  /// Get the size in Bytes of the source hashes
+  ///
+  /// If the eszip has an explicit digest size, returns that. Otherwise, returns
+  /// the default digest size of the [`Self::checksum`]. If the eszip
+  /// does not have either, returns `None`.
+  fn checksum_size(self) -> Option<u8> {
+    self
+      .checksum_size
+      .or_else(|| Some(self.checksum?.digest_size()))
+  }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[repr(u8)]
+pub enum Checksum {
+  NoChecksum = 0,
+  #[cfg(feature = "sha256")]
```

Comment:
Perhaps we should always compile with the enum variants and instead just error when a cargo feature is not enabled when serializing or deserializing with a variant that doesn't have its feature enabled? For example, when reading an eszip that is crc32 we could error saying to enable the `crc32` cargo feature in order to read that eszip.

Reply:
See https://github.com/denoland/eszip/pull/181#discussion_r1601225605